Eslint error for @computed used with single dependent key

When I need to depend on only one property like this:
```js
  @alias('service.someProperty')
  selectedItems;

  @computed('selectedItems')
  get total() {
    let total = 0;
    this.selectedItems.forEach(item => {
      total += item.price;
    });
    return total;
  }
```
I get a eslint error:
```
.../index/controller.js
  16:4  error  Do not use Computed Properties without dependent keys. If you want the property to be set only once, initialize it in init(). Please see following guide for more information: https://github.com/ember-best-practices/eslint-plugin-ember-best-practices/blob/master/guides/rules/require-dependent-keys.md  ember-best-practices/require-dependent-keys
```

which is somewhat strange. 

Opening this here as suggested in ember-decorators/ember-decorators/issues/397

--- a/lib/rules/require-dependent-keys.js
+++ b/lib/rules/require-dependent-keys.js
@@ -5,8 +5,18 @@
 const { get } = require('../utils/get');
 const MESSAGE = 'Do not use Computed Properties without dependent keys. If you want the property to be set only once, initialize it in init(). Please see following guide for more information: https://github.com/ember-best-practices/eslint-plugin-ember-best-practices/blob/master/guides/rules/require-dependent-keys.md';
 
-function isMissingDependentKeys(name, node, context) {
-  if (name === 'computed' && node.arguments.length === 1) {
+function isMissingDependentKeys(node, context) {
+  if (
+    (
+      node.arguments
+      && node.arguments.length
+      && node.arguments[0].type === 'FunctionExpression'
+    ) || (
+      node.arguments
+      && node.arguments.length
+      && node.arguments[0].type === 'ObjectExpression'
+    )
+  ) {
     context.report(node, MESSAGE);
   }
 }
@@ -24,7 +34,9 @@ module.exports = {
     return {
       CallExpression(node) {
         let name = get(node, 'callee.name') || get(node, 'callee.property.name');
-        isMissingDependentKeys(name, node, context);
+        if (name === 'computed') {
+          isMissingDependentKeys(node, context);
+        }
       }
     };
   }

--- a/package.json
+++ b/package.json
@@ -25,6 +25,7 @@
     "eslint-plugin"
   ],
   "devDependencies": {
+    "babel-eslint": "^10.0.2",
     "coveralls": "^2.13.1",
     "eslint": "^4.11.0",
     "mocha": "^3.2.0",

--- a/tests/lib/rules/require-dependent-keys.js
+++ b/tests/lib/rules/require-dependent-keys.js
@@ -1,10 +1,15 @@
 const rule = require('../../../lib/rules/require-dependent-keys');
 const MESSAGE = rule.meta.message;
 const RuleTester = require('eslint').RuleTester;
+
 const ruleTester = new RuleTester({
+  parser: 'babel-eslint',
   parserOptions: {
     ecmaVersion: 6,
-    sourceType: 'module'
+    sourceType: 'module',
+    ecmaFeatures: {
+      legacyDecorators: true
+    }
   }
 });
 
@@ -17,6 +22,33 @@ ruleTester.run('require-dependent-keys', rule, {
             return this.get('bar') * 2;
           })
         });`
+    },
+    {
+      code: `
+        export default Ember.Component({
+          foo: computed('bar', function() {
+            return this.get('bar') * 2;
+          })
+        });`
+    },
+    {
+      code: `
+        export default Ember.Component({
+          foo: computed('bar', {
+            get() {
+              return this.get('bar') * 2;
+            }
+          })
+        });`
+    },
+    {
+      code: `
+        export default class extends Ember.Object {
+          @computed('bar')
+          get foo() {
+            return this.get('bar') * 2;
+          }
+        }`
     }
   ],
   invalid: [
@@ -36,6 +68,19 @@ ruleTester.run('require-dependent-keys', rule, {
         export default Ember.Component({
           foo: computed(function() {
             return this.get('bar') * 2;
+          })
+        });`,
+      errors: [{
+        message: MESSAGE
+      }]
+    },
+    {
+      code: `
+        export default Ember.Component({
+          foo: computed({
+            get() {
+              return this.get('bar') * 2;
+            }
           })
         });`,
       errors: [{

